svm/src/account_loader.rs

```diff
@@ -160,34 +160,32 @@ impl<'a, CB: TransactionProcessingCallback> AccountLoader<'a, CB> {
         }
 
         let account = if let Some(account) = self.account_cache.get(account_key) {
-            // Inspect the account prior to collecting rent, since
-            // rent collection can modify the account.
-            self.callbacks
-                .inspect_account(account_key, AccountState::Alive(account), is_writable);
-
             // If lamports is 0, a previous transaction deallocated this account.
-            // Return None without inspecting, so it can be recreated.
+            // We must shadow the cache entry so the account may be reopened.
```

Comment:
i reworded it to be clearer, i just meant we have to ignore whats in `account_cache` because `load_transaction_account` needs to run the same logic as if there was nothing in accounts-db